cirq-core/cirq/circuits/circuit_test.py

```diff
@@ -224,6 +224,35 @@ def test_append_single():
     assert c == cirq.Circuit([cirq.Moment([cirq.X(a)])])
 
 
+def test_append_control_key():
```

Comment:
Add a test (and implementation) for subcircuits too while you're here. This should be a small and simple addition.